application/src/main/java/org/togetherjava/tjbot/commands/reminder/RemindRoutine.java

```diff
@@ -0,0 +1,142 @@
+    }
+
+    @Override
+    public @NotNull Schedule createSchedule() {
+        return new Schedule(ScheduleMode.FIXED_RATE, 0, SCHEDULE_INTERVAL_SECONDS,
+                TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void runRoutine(@NotNull JDA jda) {
+        Instant now = Instant.now();
+        database.write(context -> context.selectFrom(PENDING_REMINDERS)
+            .where(PENDING_REMINDERS.REMIND_AT.lessOrEqual(now))
+            .stream()
+            .forEach(pendingReminder -> {
+                sendReminder(jda, pendingReminder.getId(), pendingReminder.getChannelId(),
+                        pendingReminder.getAuthorId(), pendingReminder.getContent(),
+                        pendingReminder.getCreatedAt());
+
+                pendingReminder.delete();
+            }));
+    }
+
+    private static void sendReminder(@NotNull JDA jda, long id, long channelId, long authorId,
+            @NotNull CharSequence content, @NotNull TemporalAccessor createdAt) {
+        RestAction<ReminderRoute> route = computeReminderRoute(jda, channelId, authorId);
+        sendReminderViaRoute(route, id, content, createdAt);
+    }
+
+    private static RestAction<ReminderRoute> computeReminderRoute(@NotNull JDA jda, long channelId,
+            long authorId) {
+        // If guild channel can still be found, send there
+        TextChannel channel = jda.getTextChannelById(channelId);
+        if (channel != null) {
+            return createGuildReminderRoute(jda, authorId, channel);
+        }
+
+        // Otherwise, attempt to DM the user directly
+        return createDmReminderRoute(jda, authorId);
+    }
+
+    private static @NotNull RestAction<ReminderRoute> createGuildReminderRoute(@NotNull JDA jda,
+            long authorId, @NotNull TextChannel channel) {
+        return jda.retrieveUserById(authorId)
+            .onErrorMap(error -> null)
+            .map(author -> ReminderRoute.toPublic(channel, author));
+    }
+
+    private static @NotNull RestAction<ReminderRoute> createDmReminderRoute(@NotNull JDA jda,
+            long authorId) {
+        return jda.openPrivateChannelById(authorId)
+            .map(channel -> ReminderRoute.toPrivate(channel, channel.getUser()));
+    }
+
+    private static void sendReminderViaRoute(@NotNull RestAction<ReminderRoute> routeAction,
+            long id, @NotNull CharSequence content, @NotNull TemporalAccessor createdAt) {
+        Function<ReminderRoute, MessageAction> sendMessage = route -> route.channel
+            .sendMessageEmbeds(createReminderEmbed(content, createdAt, route.target()))
+            .content(route.description());
+
+        Consumer<Throwable> logFailure = failure -> logger.warn(
+                "Failed to send a reminder (id '{}'), skipping it. This can be due to a network issue,"
+                        + " but also happen if the bot disconnected from the target guild and the"
+                        + " user has disabled DMs or has been deleted.",
```

Comment:
Textblock?

Reply:
No, cause then I would have to add newlines and it shouldnt have newlines.

Reply:
Unless I use `\`.
```java
"""foo \
bar \
baz"""
```
what do u think?

Reply:
> No, cause then I would have to add newlines and it shouldnt have newlines.

Not sure what I'd do then?

Reply:
My point is, I can write it like that:
```java
"""Failed to send a reminder (id '{}'), skipping it. This can be due to a network issue, \
but also happen if the bot disconnected from the target guild and the \
user has disabled DMs or has been deleted."""
```
It uses the `\` syntax to get into the next line without actually having a newline in the text.

Some people might be familiar with this syntax (in particular console people), some not.

If you are okay with that, I will add it. I think its pretty much as-good/as-bad as using `+` 😆

Reply:
I know, I'm just not sure what I'd do.

I'm fine with either, so you can stick to this.

Reply:
I like it. I think ill try it out. Thanks.